starter-code/js/index.js

```diff
@@ -1,4 +1,92 @@
-function deleteItem(e){
+'use strict';
+
+function productTotal (){
+  console.log('Calculate prices CLICKED');
+  var pricesArray = htmlCollectionToInnerHTMLArray('unit-cost');
+  var multiplesArray = htmlCollectionToValueArray('qty-input');
+  var grandTotal = 0.00;
+    for (var j = 0; j < pricesArray.length; j++){
+      var productTotal = (pricesArray[j]) * (multiplesArray[j]);
+      document.getElementsByClassName('cost-output')[j].innerHTML = productTotal;
+      grandTotal = grandTotal + productTotal;
+    }
+  document.getElementById('grand-total').innerHTML = '$' + grandTotal; 
+  console.log(pricesArray);
+  console.log(multiplesArray);
+  console.log(grandTotal); 
+}
+
+function createNewItem (){
+  console.log('createNewItem CLICKED');//tether- to do
+}
+
+function deleteItem (id) {
+  //var i = 0;
+  //console.log(document.getElementsByClassName('prod-wrapper')[i]);
+  document.body.removeChild(document.getElementById(id));
+  console.log('deleteItem CLICKED');
+
+  //But what if we delete a middle product? Let's make the deleteItems params go in order!
+  var ProdsRemaining = document.getElementsByClassName('prod-wrapper').length;
+  var newButton = '';
+  for (var i = 0; i < ProdsRemaining; i++){
+    document.getElementsByClassName('prod-wrapper')[i].id = i;
+    newButton = '<button onclick="deleteItem(\'' + i + '\')" class="btn btn-delete" type="button">Delete</button>';
+    document.getElementsByClassName('delete-box')[i].innerHTML = newButton;
+  }
+  
+  //LOL!!! I didn't read this sentence in the instructions:
+  //"Use e.currentTarget to access the "Delete" button that 
```

Comment:
Your problem it will easier resolve with the `e.currentTarget`. 
But good job! 👍